src/server/mom_info.c

```diff
@@ -503,6 +505,209 @@ delete_svrmom_entry(mominfo_t *pmom)
+ *	a vnode pool.  Create the pool if need be (not yet exists).
+ *
+ * @param[in] pmom - Pointer to the mominfo_t for the Mom
+ * @return Error code
+ * @retval - 0 - Success
+ * @retval - pbs_errno - Failure code
+ *
+ * @par MT-safe: No
+ */
+int
+add_mom_to_pool(mominfo_t *pmom)
+{
+	vnpool_mom_t *ppool;
+	mom_svrinfo_t *psvrmom = (mom_svrinfo_t *)pmom->mi_data;
+	int		added_pool = 0;
+	int		i;
+	mominfo_t    **tmplst;
+
+	if (psvrmom->msr_vnode_pool == 0)
+		return PBSE_NONE;	/* Mom not in a pool */
+
+	ppool = find_vnode_pool(pmom);
+	if (ppool != NULL) {
+		/* Found existing pool. Is Mom already in it? */
+		for (i = 0; i < ppool->vnpm_nummoms; ++i) {
+			if (ppool->vnpm_moms[i] == pmom) {
+				sprintf(log_buffer, "POOL: add_mom_to_pool - "
+					"Mom already in pool %ld",
+					psvrmom->msr_vnode_pool);
+				log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_NODE,
+					LOG_INFO, pmom->mi_host,log_buffer);
+				return PBSE_NONE; /* she is already there */
+			}
+		}
+	}
+
+	/* The pool doesn't exist yet, we need to add a pool entry */
+	if (ppool == NULL) {
+		ppool = (vnpool_mom_t *)calloc(1, (size_t)sizeof(struct vnpool_mom));
+		if (ppool == NULL) {
+			/* no memory */
+			log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_NODE, LOG_ERR,
+				  pmom->mi_host,
+				  "Failed to expand vnode_pool_mom_list");
+			return PBSE_SYSTEM;
+		}
+		added_pool = 1;
+		ppool->vnpm_vnode_pool = psvrmom->msr_vnode_pool;
+	}
+
+	/* now add Mom to pool list, expanding list if need be */
+
+	/* expand the array, perhaps from nothingness */
+	tmplst = (mominfo_t **)realloc(ppool->vnpm_moms, (ppool->vnpm_nummoms+1)*sizeof(mominfo_t *));
+	if (tmplst == NULL) {
+		log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_NODE,
+			  LOG_ERR, pmom->mi_host,
+			  "unable to add mom to pool, no memory");
+
+		if (added_pool)
+			free(ppool);
+		return PBSE_SYSTEM;
+	}
+	ppool->vnpm_moms = tmplst;
+	ppool->vnpm_moms[ppool->vnpm_nummoms++] = pmom;
+
+	sprintf(log_buffer, "Mom %s added to vnode_pool %ld", pmom->mi_host, psvrmom->msr_vnode_pool);
+	log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, msg_daemonname, log_buffer);
+	if (ppool->vnpm_inventory_mom == NULL) {
+		ppool->vnpm_inventory_mom = pmom;
+		psvrmom->msr_has_inventory = 1;
+		sprintf(log_buffer, msg_new_inventory_mom, psvrmom->msr_vnode_pool, pmom->mi_host);
+		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_DEBUG, msg_daemonname, log_buffer);
+	}
+
+
+	if (vnode_pool_mom_list == NULL) {
+		vnode_pool_mom_list = ppool;
+	} else if (added_pool == 1) {
+		ppool->vnpm_next = vnode_pool_mom_list;
+		vnode_pool_mom_list = ppool;
+	}
+	return PBSE_NONE;
+}
+
+/**
+ * @brief
+ *	remove a Mom (mominfo_t) from the list of Moms associated with managing
+ *	a vnode pool.
+ *
+ * @param[in] pmom - Pointer to the mominfo_t for the Mom
+ *
+ * @par MT-safe: No
+ */
+void remove_mom_from_pool(mominfo_t *pmom)
+{
+	vnpool_mom_t *ppool;
+	mom_svrinfo_t *psvrmom = (mom_svrinfo_t *)pmom->mi_data;
+	int			   i;
```

Comment:
same here, and add space around `+` in one of the `for` below

Reply:
I found it more readable in this case without the spaces as it appears within the for. I've made changes to the rest of the places. Hope you agrees!